test/controllers/sessions_controller_test.rb

```diff
@@ -257,6 +262,15 @@
         @controller.send(:current_user).must_equal(user)
       end
 
+      it 'logs the user in with USE_LDAP_UID_AS_EXTERNAL_ID' do
+        Rails.application.config.samson.ldap.stub(:uid, 'sAMAccountName') do
+          with_env 'USE_LDAP_UID_AS_EXTERNAL_ID' => 'true' do
+            post :ldap
+            @controller.send(:current_user).must_equal(user)
```

Comment:
this would still pass if the new code was removed, right ?
... so add something to make sure it does not use the old code